File: minwook/4주차/자릿수_더하기.java
public class Solution {
    public int solution(int n) {
        int answer = 0;
        int num = 0;
        int digit = 0;
        
        for(int i=10; i<=n; i*=10){ // i * 10으로 나눈 나머지 값 계속 덧셈 (첫번째 수는 덧셈 x)
            num = n%i; 
            n -= num;
            while(num>=10) // 자릿수 값을 위해 10보다 크거나 같을 경우 나누기
                num/=10;
            answer += num;
            digit++; // 자릿수 확인을 위한 숫자
        }

        return answer + n/(int)Math.pow(10,digit); // n의 첫번째 수 덧셈
    }
}

// while문을 사용하면 더 간단하게 작성 가능

File: yongkeun/4week/자릿수더하기.js
function solution(n) {
  let answer = 0;
  String(n)
    .split("")
    .forEach((value) => (answer += Number(value))); //n을 문자열로 바꾼 후 split을 이용해 하나하나 분리하고, 다시 숫자로 바꾼 후 answer에 더함
  return answer;
}
